src/main/InternalProcess.js

```diff
@@ -123,13 +215,13 @@ class InternalProcess {
       this.messageQueue = [];
       const pid = this.process.pid;
 
-      logger.info(`ending process ${pid} ...`);
-      console.log(`ending process ${pid} ...`);
+      logger.info(`ending process ${pid} ...`, logType);
       this.active = false;
-      this.process.once("exit", () => {
-        resolve();
-      });
-      this.process.disconnect();
+      this.process &&
```

Comment:
same here you could have `const {process} = this` at the beginning of the function to prove to flow only once.